e2e/glustershd_test.go

```diff
@@ -0,0 +1,60 @@
+package e2e
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/gluster/glusterd2/pkg/api"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGlusterShd(t *testing.T) {
+
+	r := require.New(t)
+
+	gds, err := setupCluster("./config/1.yaml", "./config/2.yaml")
+	r.Nil(err)
+	defer teardownCluster(gds)
+
+	brickDir, err := ioutil.TempDir("", "TestShdEnable")
+	r.Nil(err)
+	defer os.RemoveAll(brickDir)
+
+	var brickPaths []string
+	for i := 1; i <= 4; i++ {
+		brickPath, err := ioutil.TempDir(brickDir, "brick")
+		r.Nil(err)
+		brickPaths = append(brickPaths, brickPath)
+	}
+
+	client := initRestclient(gds[0].ClientAddress)
+	volname1 := "testvol1"
+	reqVol := api.VolCreateReq{
+		Name: volname1,
+		Subvols: []api.SubvolReq{
+			{
+				ReplicaCount: 2,
+				Type:         "replicate",
+				Bricks: []api.BrickReq{
+					{NodeID: gds[0].PeerID(), Path: brickPaths[0]},
+					{NodeID: gds[1].PeerID(), Path: brickPaths[1]},
+				},
+			},
+		},
+		Force: true,
+	}
+	vol1, err := client.VolumeCreate(reqVol)
+	r.Nil(err)
+
+	err = client.GlusterShdEnable(vol1.Name)
```

Comment:
Don't you have to start the volume (may be mount too ?) and then enable/disable self heal ? Please check.

Reply:
No, I guess I don't need to because this API will be called just after volume creation success.

Reply:
I can see that it is called so. What I meant was - to consider that self heal daemon start and stop is working, should we test it with the volume in started state or not ? IOW, how do normal users usually enable and disable self heal daemon ?

Reply:
I am adding a check for volume state whether it's in started state or not.